### DISCUSSION TITLE
Halftone code turned to nodes

### DISCUSSION THREAD

**Discussion by JPLost** [bnpr/Malt#122] [Showcase]
Hello,

I'm converting the halftone code to nodes to see if there are problems regarding the node setup.
This is actually quite interesting thing to see how well it functions.

Attached you will find zipped blender file with Suzanne mesh with the halftone nodetree. This file has been made with Blender 2.93.4
[Halftone_Nodes.zip](https://github.com/bnpr/Malt/files/7149970/Halftone_Nodes.zip)

I noticed that conversion of ivec2 and bool are apparently not available?

**Comment by pragma37** [comment]
> I noticed that conversion of ivec2 and bool are apparently not available?

There are some bool nodes in *Node Utils - bool*, and all basic types have an *if_else* node.

I'd like to add automatic type conversion to nodes, so you can simply link an ivec2 to vec2 sockets.
For now, you can always fall back to the *Inline Code* node.

![imagen](https://user-images.githubusercontent.com/36610999/133003735-26991569-2799-46ac-a3f7-d26559472b1f.png)

BTW, I'm not sure what's the rationale for multiplying *get_diffuse* by *scene_diffuse*, but to use *scene_diffuse* you should always link a position (*surface_position*) and a normal input (*get_normal*).

**Reply by JPLost** [reply]
Let's drop the revised file here.
[Halftone_Nodes_Rev.zip](https://github.com/bnpr/Malt/files/7155827/Halftone_Nodes_Rev.zip)
